### ISSUE TITLE
Bug in borer-derivation? deriveCodec fails to compile when using type alias

### ISSUE THREAD

**Issue by MarconZet** [sirthias/borer#388]
```
type Foo = String
case class Bar(bar: Foo)
implicit val barCodec: Codec[Bar] = deriveCodec
```
fails build with:
```
[error] MyBorerAkkaSerializer.scala:54:39: value writeMyBorerAkkaSerializer.this.Foo is not a member of io.bullet.borer.Writer
[error]   implicit val barCodec: Codec[Bar] = deriveCodec
```
Expected behaviour: Build successful

I found a workaround with:
```
implicit val barCodec: Codec[Bar] = Codec(
  Encoder.from(Bar.unapply _),
  Decoder.from(Bar.apply _)
)
```
but that's far from an elegant solution.

**Comment by sirthias** [comment]
Ok, thank you for reporting.
I happen to have time right now to look at this and will do so right away...

**Comment by sirthias** [comment]
Thank you, @MarconZet, for reporting.
The fix is already in the master branch.
Would you like a quick patch release to unblock this issue on your side?

**Comment by PawelLipski** [comment]
We'd appreciate a patch release :)

**Comment by sirthias** [comment]
Ok, just pushed borer 1.7.1 to sonatype.
You can pull it from there or wait for the sync to maven central, which is probably going to take a few hours though...
Thanks again for reporting!

**Comment by PawelLipski** [comment]
Wow that's one of the fastest report-to-release cycles I've seen in OSS so far ❤️

**Comment by sirthias** [comment]
:)
It was just a lucky coincidence that the bug report hit a perfect break between things on my side.
I'm glad that borer works out for your setup!

**Comment by PawelLipski** [comment]
Actually, FYI, as you've probably noticed, we're working on ready-to-use serializer for Akka messages/event/states, since Jackson that's suggested by Akka docs is a joke (surprising runtime failures, no support for Scala `objects`/ADTs etc.) 😅

Are you aware of any work that has been already done in that direction? There's borer-akka-compat but AFAICS it doesn't provide a ready Serializer, just Codecs for ActorRefs etc.

**Comment by PawelLipski** [comment]
Also @sirthias ... would it be possible to retain support for Scala 2.12 in 1.7+? Since AFAICS 1.7.0 and 1.7.1 have only been published for 2.13 ://

**Comment by sirthias** [comment]
> There's borer-akka-compat but AFAICS it doesn't provide a ready Serializer, just Codecs for ActorRefs etc.

Yes. The reason is that we haven't needed these things ourselves yet.
I'd be happy to review and merge any PRs if you want to submit any...

**Comment by PawelLipski** [comment]
Great! We're prototyping that as a separate project for convenience as for now, but it makes sense to integrate it back into main borer repo... opened https://github.com/VirtusLab/akka-safer-serializer/issues/18 for that

**Comment by mushtaq** [comment]
@PawelLipski We are using borer for akka-serialization using the pattern mentioned [here](https://github.com/sirthias/borer/issues/161#issuecomment-581257120).

**Comment by PawelLipski** [comment]
@mushtaq actually! since you're using Borer for a bit longer than us... do you have any experiences with **migrating Akka events** when Borer is used as a serializer (or any other library, for that matter)?
